src/Simplex/Chat/Store/Messages.hs

```diff
@@ -1882,11 +1882,12 @@ deleteGroupChatItem db User {userId} g@GroupInfo {groupId} ci = do
       WHERE user_id = ? AND group_id = ? AND chat_item_id = ?
     |]
     (userId, groupId, itemId)
+  pure item
 
-updateGroupChatItemModerated :: forall d. MsgDirectionI d => DB.Connection -> User -> GroupInfo -> ChatItem 'CTGroup d -> GroupMember -> UTCTime -> IO (ChatItem 'CTGroup d)
-updateGroupChatItemModerated db User {userId} GroupInfo {groupId} ci m@GroupMember {groupMemberId} deletedTs = do
+updateGroupChatItemModerated :: DB.Connection -> User -> GroupInfo -> GroupMember -> UTCTime -> CChatItem 'CTGroup -> IO (CChatItem 'CTGroup, CChatItem 'CTGroup)
```

Comment:
this change seems not necessary

Reply:
also, why change the order of params even if it is necessary

Reply:
and why does it return its own parameter unchanged?

Reply:
unwrapping of existencial wrappers can always happen outside, there is _never_ a reason to bring it inside, and here we made the type of this function less precise (e.g., it can change item direction).

Functions that return existencial wrappers are only necessary if we don't know which type it returns (e.g., if we pass ID and we are expecting any direction for this ID), in this case we do know - it's the same type we passed (which is the restriction that was lost in type with this change).